pallets/compliance-manager/src/lib.rs

```diff
@@ -785,39 +989,59 @@ impl<T: Trait> ComplianceManagerTrait<T::Balance> for Module<T> {
                 weight_for::weight_for_reading_asset_compliance::<T>(),
             ));
         }
-        for requirement in asset_compliance.requirements {
-            if let Some(from_did) = from_did_opt {
-                requirement_count += requirement.sender_conditions.len();
-                if !Self::are_all_conditions_satisfied(
+
+        // Check for whether sender & has the scope claim or not if not then fail the txn.
+        // To optimize we are not checking it again and again with the respective conditions, it
+        // gets checked only once and if it is valid then its result is tied up with the conditions result.
+        //
+        // Note - Due to this check `ConditionType::HasValidProofOfInvestor` is an implicit transfer condition and it only
+        // lookup for the claims those are provided by the trusted_claim_issuers.
+        let is_requirements_needed_to_check = match Self::implicit_requirements_status(ticker) {
```

Comment:
Instead of having this var, it's cleaner to just return early in case of failures.
something like
```
if implicit_requirements_status && !is_sender_and_receiver_has_valid_scope_claim {
    return;
}
```

Reply:
I don't want to compute the validity of the scope claim for every ticker, with current logic it will only compute when it is needed.

Reply:
>I don't want to compute the validity of the scope claim for every ticker

I don't see how that is relevant here. You should always only verify the scope for the ticker of the token that is being transferred.

Reply:
Whenever the scope claim is activated, the function needs to verify whether the sender and receiver have the valid scope claim (first generate dynamically and then verify). If I add this call something like this 

```
if implicit_requirements_status() && !is_sender_and_receiver_has_valid_scope_claim() {
    return;
}
```
then even if the implicit_requirement_status is not activated for the give ticker it will create a dynamic condition and then lookup for the scope claim, this needs computation that I want to avoid.

Reply:
>even if the implicit_requirement_status is not activated for the give ticker it will create a dynamic condition and then lookup for the scope claim, this needs computation that I want to avoid.

No, `&&` is short-circuiting. i.e. If the first condition returns false, the second will not be checked. That means, if `implicit_requirement_status` is not activated, `is_sender_and_receiver_has_valid_scope_claim` will not be triggered.